api/v1beta1/azuremachine_default_test.go

```diff
@@ -401,39 +402,102 @@ func TestAzureMachineSpec_SetNetworkInterfacesDefaults(t *testing.T) {
 	}
 }
 
+func TestAzureMachineSpec_GetOwnerCluster(t *testing.T) {
+	g := NewWithT(t)
+
+	tests := []struct {
+		name            string
+		maxAttempts     int
+		wantedName      string
+		wantedNamespace string
+		wantErr         bool
+	}{
+		{
+			name:            "ownerCluster is returned",
+			maxAttempts:     1,
+			wantedName:      "test-cluster",
+			wantedNamespace: "default",
+			wantErr:         false,
+		},
+		{
+			name:            "ownerCluster is returned after 2 attempts",
+			maxAttempts:     2,
+			wantedName:      "test-cluster",
+			wantedNamespace: "default",
+			wantErr:         false,
+		},
+		{
+			name:            "ownerCluster is not returned after 5 attempts",
+			maxAttempts:     5,
+			wantedName:      "test-cluster",
+			wantedNamespace: "default",
+			wantErr:         true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			client := mockClient{ReturnError: tc.wantErr}
```

Comment:
nit:
```suggestion
			g := NewWithT(t)
			client := mockClient{ReturnError: tc.wantErr}
```

Reply:
Is this the recommended pattern? I see that most other tests only declare `g := NewWithT(t)` at the start of the test function.

Reply:
If so, I might be more in favor of keeping things consistent in this PR, and creating a separate PR to make this change across the package.

Reply:
From my point of view, that would be the recommended pattern. When the `g` is created inside the subtest, error messages will include the name of the exact subtest that failed. Otherwise it's not always obvious exactly which test case failed.